### ISSUE TITLE
FR: data/aws_cloudwatch_log_group

### ISSUE THREAD

**Issue by jkroepke** [terraform-providers/terraform-provider-aws#3247]
Hi there,

Thank you for opening an issue. Please note that we try to keep the Terraform issue tracker reserved for bug reports and feature requests. For general usage questions, please see: https://www.terraform.io/community.html.

### Terraform Version
Terraform v0.11.3
+ provider.aws v1.8.0
+ provider.local v1.0.0
+ provider.null v1.0.0
+ provider.template v1.0.0

### Affected Resource(s)
Please list the resources as a list, for example:
- d/aws_cloudwatch_log_group

Hi,

For the API Gateway, amazon create automaticly Cloudwatch Log Groups based on the aws gw id (e.g.)

I want to import the Log Group and apply additional configurations. Thats why i'm need the aws_cloudwatch_log_group data source.

**Comment by bflad** [comment]
Hi @jkroepke, you specifically are mentioning the following:

> I want to import the Log Group and apply additional configurations.

In Terraform's configuration model this is handled by a resource, not a data source. Think of data sources as read-only information about AWS itself or your infrastructure.

The documentation for the existing [`aws_cloudwatch_log_group` resource](https://www.terraform.io/docs/providers/aws/r/cloudwatch_log_group.html) has information on importing an existing CloudWatch log group into Terraform.

With an example configuration such as:
```hcl
resource "aws_api_gateway_rest_api" "example" {
  name = "example"
  # ... other configuration ...
}

resource "aws_api_gateway_stage" "example" {
  stage_name = "example"
  rest_api_id = "${aws_api_gateway_rest_api.example.id}"
  # ... other configuration ...
}

# This should work a configuration when importing (it if it exists), however could have issues if being created the same time as above
resource "aws_cloudwatch_log_group" "example" {
  name = "API-Gateway-Execution-Logs_${aws_api_gateway_rest_api.example.id}/${aws_api_gateway_stage.example.stage_name}"
  # ... other configuration ...
}
```

You might be have some trouble doing this automatically when the API Gateway is created at the same time as the CloudWatch log group though. AWS might create the log group automatically before Terraform does, in which Terraform will throw an error on creating the log group. If that is your situation, please let us know.

**Comment by bflad** [comment]
The `aws_cloudwatch_log_group` data source has been released in [version 1.15.0 of the AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#1150-april-18-2018). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.
